src/main/kotlin/blackjack/domain/player/Players.kt

```diff
@@ -12,10 +15,18 @@ data class Players(val players: List<Player>) : List<Player> by players {
 
     init {
         require(players.size >= MINIMUM_GAMER)
+
+        if (players.count { it is Dealer } > DEALER_COUNT) {
+            throw IllegalStateException(DEALER_ALREADY_EXIST)
+        }
```

Comment:
```suggestion
        if (players.count { it is Dealer } == DEALER_COUNT) {
            throw IllegalStateException(DEALER_ALREADY_EXIST)
        }
```
조건이 딜러가 한명인게 좋을거같아요.

Reply:
``` kotlin
require (players.count { it is Dealer } == DEALER_COUNT)
```
를 의도하신 것이 맞을까요?

Reply:
작성한 소스를확인했는데, players는 게임에 참가하는 플레이어(딜러, 게이머)를 의도했기 때문에 
게이머만으로도 목록이 생성이 되어야합니다. `딜러는 한명이 되어야한다`는 조건을 만족하기 위해 
``` kotlin
if (players.count { it is Dealer } > DEALER_COUNT) {
            throw IllegalStateException(DEALER_ALREADY_EXIST)
}
```
을 유지해야할 것같아요

Reply:
답변이늦었습니다.
require (players.count { it is Dealer } == DEALER_COUNT)
```
       require (players.count { it is Dealer } == DEALER_COUNT)

```
이걸 의도한게 맞습니다.


if (players.count { it is Dealer } > DEALER_COUNT) {
            throw IllegalStateException(DEALER_ALREADY_EXIST)
}
이조건이면 딜러가 두명 이상일경우는 유효성 체크가 되지  못하는게 아닐까요?

Reply:
딜러가 두명 이상 발생할 수 있는 케이스에 대해서는 테스트 코드를 작성해보았습니다.
혹시 오류가 생길 수 있는 케이스가 더 있을까요? players로 묶다보니 리뷰어님이 제안해주신 코드를 반영하지 못해 아쉽네요.

``` kotlin
@Test
    fun `플레이어 목록에 딜러가 2명이면 예외를 던진다`() {
        val dealer1 = Dealer.of()
        val dealer2 = Dealer.of()

        assertThrows<IllegalStateException> { Players(listOf(dealer1, dealer2)) }
    }

    @Test
    fun `플레이어 목록에 딜러가 2명이면 예외를 던진다2`() {
        val profile1 = Profile.from(Name("player1"))
        val givenGamer1 = Gamer(profile1, Cards(emptyList()))
        val dealer1 = Dealer.of()
        val dealer2 = Dealer.of()

        val players = Players(listOf(givenGamer1, dealer1))
        assertThrows<IllegalStateException> { players.addPlayer(dealer2) }
    }
```